IPNS: context deadline exceeded

I was excited to read that 0.5 speeds up IPNS. So I wanted to give it a spin. Unfortunately I get an error now:

```
ligi@komputing:~$ ipfs add -Q -r /opt/source-verify/production/source-verify/repository/
QmfXZmzSaEZvEWrJtTjJkmXDV5Dpoxz2diHFPHvnsrexa3
ligi@komputing:~$ ipfs name publish QmfXZmzSaEZvEWrJtTjJkmXDV5Dpoxz2diHFPHvnsrexa3
Error: context deadline exceeded
```
running this version:

```
go-ipfs version: 0.5.0
Repo version: 9
System version: amd64/linux
Golang version: go1.13.10

```

When I encounter this problem, the daemon logs a "error when determining the last published IPNS record" (using `ipfs log level namesys debug`). It would be nice if CLI client shows that message instead of the obscure one.

```
Apr 29 20:11:19 localhost ipfs[342]: 2020-04-29T20:11:19.383+0900        DEBUG        namesys        namesys/publisher.go:126        error when determining the last published IPNS record for Qmak3N3ey6AAGndq5zwZD3N2oP5gJkS4A8QGEqVUR7J7gT: context deadline exceeded
Apr 29 20:11:19 localhost ipfs[342]: [153B blob data]
Apr 29 20:11:19 localhost ipfs[342]: [159B blob data]
```

--- a/namesys/publisher.go
+++ b/namesys/publisher.go
@@ -22,7 +22,6 @@ import (
 
 const ipnsPrefix = "/ipns/"
 
-const PublishPutValTimeout = time.Minute
 const DefaultRecordEOL = 24 * time.Hour
 
 // IpnsPublisher is capable of publishing and resolving names to the IPFS
@@ -269,15 +268,10 @@ func PublishPublicKey(ctx context.Context, r routing.ValueStore, k string, pubk 
 	}
 
 	// Store associated public key
-	timectx, cancel := context.WithTimeout(ctx, PublishPutValTimeout)
-	defer cancel()
-	return r.PutValue(timectx, k, pkbytes)
+	return r.PutValue(ctx, k, pkbytes)
 }
 
 func PublishEntry(ctx context.Context, r routing.ValueStore, ipnskey string, rec *pb.IpnsEntry) error {
-	timectx, cancel := context.WithTimeout(ctx, PublishPutValTimeout)
-	defer cancel()
-
 	data, err := proto.Marshal(rec)
 	if err != nil {
 		return err
@@ -285,7 +279,7 @@ func PublishEntry(ctx context.Context, r routing.ValueStore, ipnskey string, rec
 
 	log.Debugf("Storing ipns entry at: %s", ipnskey)
 	// Store ipns entry at "/ipns/"+h(pubkey)
-	return r.PutValue(timectx, ipnskey, data)
+	return r.PutValue(ctx, ipnskey, data)
 }
 
 // InitializeKeyspace sets the ipns record for the given key to

